src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -12,82 +12,101 @@
 namespace Microsoft.ML.Data
 {
     /// <summary>
-    /// Attach to a member of a class to indicate that the item type should be of class key.
+    /// Allow member to be marked as a <see cref="KeyType"/>.
     /// </summary>
+    /// <remarks>
+    /// Can be applied only for member of following types: <see cref="byte"/>, <see cref="ushort"/>, <see cref="uint"/>, <see cref="ulong"/>
+    /// </remarks>
     [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
     public sealed class KeyTypeAttribute : Attribute
     {
-        // REVIEW: Property based, but should I just have a constructor?
+        /// <summary>
+        /// Marks member as <see cref="KeyType"/>.
+        /// </summary>
+        /// <remarks>
+        /// Cardinality of <see cref="KeyType"/> would be maximum legal value of member type.
+        /// </remarks>
+        public KeyTypeAttribute()
+        {
+
+        }
+
+        /// <summary>
+        /// Marks member as <see cref="KeyType"/> and specifies <see cref="KeyType"/> cardinality.
+        /// </summary>
+        /// <param name="count">Size of key type.</param>
+        public KeyTypeAttribute(ulong count)
+        {
+            Count = new KeyCount(count);
```

Comment:
>Count [](start = 12, length = 5)

If you make another iteration, maybe this could be renamed to `KeyCount` to reflect its type. Not a blocking comment though.